crates/testing/src/predicates/event.rs

```diff
@@ -58,18 +58,22 @@ where
     TYPES: NodeType,
 {
     let info = format!("{:?}", events);
-    let mut set: HashSet<_> = events.into_iter().collect();
-
-    let function = move |e: &Arc<HotShotEvent<TYPES>>| match set.take(e.as_ref()) {
-        Some(_) => {
-            if set.is_empty() {
-                PredicateResult::Pass
-            } else {
-                PredicateResult::Incomplete
+
+    // Remove duplicate events.
+    let mut events = events;
+    events.dedup();
+
+    let function =
+        move |e: &Arc<HotShotEvent<TYPES>>| match events.iter().find(|x| *x == e.as_ref()) {
```

Comment:
If i understand `find` correctly, I don't think this is right: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find

It looks like it clears the iterator until it finds the first match.

Reply:
Fixed by just mapping over the events as suggested below.